react-app-drawer-component/my-app/src/AppDrawer.tsx

```diff
@@ -0,0 +1,59 @@
+import './AppDrawer.css';
+
+export type MenuItem = {
+  title: string;
+};
+
+type AppDrawerProps = {
+  menuItems: MenuItem[];
+  heading: string;
+  drawerState: boolean;
+  onMenuItemClicked: Function;
```

Comment:
`Function` is a very generic type. It matches any function with any number of arguments and any return type. It's best to be very explicit about the functions, instead.

Reading your code, it looks like this definition would be better for `onMenuItemClicked`:
`onMenuItemClicked: (item: string) => void`